components/match2/wikis/brawlstars/match_group_input_custom.lua

```diff
@@ -342,67 +188,37 @@ function mapFunctions.getExtraData(map)
 
 	local bans = {}
 	local getCharacterName = FnUtil.curry(MatchGroupInput.getCharacterName, BrawlerNames)
-	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+	for opponentIndex = 1, opponentCount do
 		bans['team' .. opponentIndex] = {}
 		for _, ban in Table.iter.pairsByPrefix(map, 't' .. opponentIndex .. 'b') do
 			ban = getCharacterName(ban)
 			table.insert(bans['team' .. opponentIndex], ban)
 		end
 	end
 
-	map.extradata.bans = Json.stringify(bans)
-	map.bestof = bestof
-
-	return map
-end
-
----@param map table
----@return table
-function mapFunctions.getScoresAndWinner(map)
-	map.score1 = tonumber(map.score1 or '')
-	map.score2 = tonumber(map.score2 or '')
-	map.scores = { map.score1, map.score2 }
-	if Table.includes(NOT_PLAYED, string.lower(map.winner or '')) then
-		map.winner = 0
-		map.resulttype = 'np'
-	elseif Logic.isNumeric(map.winner) then
-		map.winner = tonumber(map.winner)
-	end
-	local firstTo = math.ceil( map.bestof / 2 )
-	if (map.score1 or 0) >= firstTo then
-		map.winner = 1
-		map.finished = true
-	elseif (map.score2 or 0) >= firstTo then
-		map.winner = 2
-		map.finished = true
-	end
+	extradata.bans = bans
 
-	return map
+	return extradata
 end
 
 ---@param map table
+---@param opponentCount integer
 ---@return table
-function mapFunctions.getParticipantsData(map)
+function MapFunctions.getParticipants(map, opponentCount)
```

Comment:
attaches the data to wrong player in some cases
(current implementation does it the same but imo we should fix it)

Reply:
Same as the valorant PR, OOS for this, but look it afterwards